runner/src/index.ts

```diff
@@ -23,37 +23,82 @@ const KitchenSink = {
 };
 
 export function test() {
-  let input = JSON.stringify(KitchenSink);
-  let output = JSON.parse(Test.callString("reflectJsonObject", input));
+  Test.group("schema v1-draft encoding types", () => {
+    let input = JSON.stringify(KitchenSink);
+    let output: typeof KitchenSink = Test.call("reflectJsonObject", input)
+      .json();
 
-  matchIdenticalTopLevel(output);
-  matchIdenticalEmbedded(output.anEmbeddedObject);
-  output.anEmbeddedObjectArray.forEach(matchIdenticalEmbedded);
+    matchIdenticalTopLevel(output);
+    matchIdenticalEmbedded(output.anEmbeddedObject);
+    output.anEmbeddedObjectArray.forEach(matchIdenticalEmbedded);
 
-  // dates and JSON encodings between languages are a little fuzzy.
-  // so, rather than test stringified equality, we test the value of
-  // the date in various forms.
-  matchDate(output);
+    // dates and JSON encodings between languages are a little fuzzy.
+    // so, rather than test stringified equality, we test the value of
+    // the date in various forms.
+    matchDate(output);
 
-  Test.assertEqual(
-    "reflectJsonObject preserved optional field semantics",
-    output.anOptionalString,
-    KitchenSink.anOptionalString,
-  );
+    Test.assertEqual(
+      "reflectJsonObject preserved optional field semantics",
+      output.anOptionalString,
+      KitchenSink.anOptionalString,
+    );
 
-  let inputS = KitchenSink.aString;
-  let outputS = Test.callString("reflectUtf8String", inputS);
-  Test.assertEqual("reflectUtf8String preserved the string", outputS, inputS);
+    let inputS = KitchenSink.aString;
+    let outputS = Test.call("reflectUtf8String", inputS).text();
+    Test.assertEqual("reflectUtf8String preserved the string", outputS, inputS);
 
-  let inputB = (new TextEncoder()).encode(KitchenSink.aString).buffer;
-  let outputB = Test.callBuffer("reflectByteBuffer", inputB);
+    let inputB = (new TextEncoder()).encode(KitchenSink.aString).buffer;
+    let outputB = Test.call("reflectByteBuffer", inputB).arrayBuffer();
 
-  // TODO compare the bytes
-  Test.assertEqual(
-    "reflectByteBuffer preserved the buffer length",
-    outputB.byteLength,
-    inputB.byteLength,
-  );
+    // TODO compare the bytes
+    Test.assertEqual(
+      "reflectByteBuffer preserved the buffer length",
+      outputB.byteLength,
+      inputB.byteLength,
+    );
+  });
+
+  Test.group("check signature and type variations", () => {
+    // should call a the `noInputWithOutputHost` host function passing it
+    // a string "noInputWithOutputHost" which it should return
+    let noInputWithOutputOutput = Test.call(
+      "noInputWithOutput",
+      undefined,
```

Comment:
I think you might be able to remove the explicit `undefined` argument here? Not 100% sure about that though.

Reply:
I thought so too! TypeScript complains about it though, and I _think_ we needed to add `input?: Input` to the function signature for it to be less picky. I think it's actually a bit more clear what's going on here vs. omitting the `undefined` entirely.. but I could go either way

Reply:
Yeah probably not worth worrying about it

Reply:
ok will revisit if anyone asks about it